src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -25,6 +25,9 @@ public class ViewBuffer : IHtmlContentBuilder
         private readonly IViewBufferScope _bufferScope;
         private readonly string _name;
         private readonly int _pageSize;
+        private ViewBufferPage _singlePage = null;          // Limits allocation if the ViewBuffer has only one page (frequent case).
+        private List<ViewBufferPage> _multiplePages = null; // Allocated only if necessary
+        private ViewBufferPage _currentPage = null;
```

Comment:
If you're adding a `_currentPage` is there really a need for `_singlePage`? `Count` would look like:
```
get
{
    if (_multiplePages != null)
    {
        return _multiplePages.Count;
    }
    if (_currentPage != null)
    {
        return 1;
    }
    return 0;
}
```